cmd/snap/main.go

```diff
@@ -129,11 +129,19 @@ func init() {
 	}
 }
 
+var accessDeniedHelp = `Access denied.
+
+Please use "sudo" or the "snap login <userid>" command
+`
+
 func main() {
 	if err := run(); err != nil {
 		if e, ok := err.(*flags.Error); ok && e.Type == flags.ErrHelp {
 			fmt.Fprintf(Stdout, "%v\n", err)
 			os.Exit(0)
+		} else if e, ok := err.(*client.Error); ok && e.Message == "access denied" {
```

Comment:
I suggest having a Client.StatusCode and decorating it with what we get in response.StatusCode.